integr8ly/integreatly-operator
# MGDAPI-3595 add source code for scorecard test img

# Issue link
https://issues.redhat.com/browse/MGDAPI-3595

# What
* migrate RHOAM KUTTL test image config from [here](https://github.com/psturc/kuttl-scorecard-rhoam) to this repo
* create a Makefile with useful commands for preparing/running scorecard tests and building a container image

# Verification steps
1. Target OpenShift cluster with `oc`
2. If you have RHOAM installed, continue, otherwise create `redhat-rhoam-operator` namespace
3. Run
```bash
make scorecard/bundle/prepare
make scorecard/service_account/prepare
make scorecard/test/run SCORECARD_TEST_NAME=basic-check-spec-test
```
4. You should get an output with the following test result
```
      "status": {
        "results": [
          {
            "name": "basic-check-spec",
            "state": "pass",
            "creationTimestamp": null
          }
        ]
      }
```
5. Test that the scorecard test KUTTL image can be built using the make command
```bash
# The container image build command should succeed
make scorecard/build
# Get the name of the image that was build - it should have a tag quay.io/integreatly/scorecard-test-kuttl:master
docker images | grep scorecard-test-kuttl

quay.io/integreatly/scorecard-test-kuttl               master      f3d91abe34ed   22 hours ago   340MB
```
